[31] move bower_components out of build into node_modules

inpercima opened the pull request:
* move bower_components to node_modules
* refactoring Gruntfile.js
* add bowercopy to package.json
* create tasks to use bowercopy

inpercima commented:
This commit should used for discussion. What about the idea behind this? On monday ;)

inpercima commented:
@sepe81 can you please review these changes and tell us what you think? :-)

sepe81 commented on Gruntfile.js:
```diff
@@ -5,9 +5,13 @@ module.exports = function(grunt) {
+
+    src_app_js: 'src/main/resources/public/js/app.js',
+
+    src_files_js: 'src/main/resources/public/js/*.js',
```
why did you leave the /** from line 35 of the previous version?

inpercima replied:
``src_dir_js`` was ``src/main/resources/public/js`` and then in the option the ``/**`` was added. I desided to create an new variable ``src_files_js`` with include ``*.js``. In my tests it was working fine. Also subfiles were checked, but I will review this again.

inpercima replied:
I checked this and found out that this was wrong so ich add ``/**`` again. Also removed the ``*`` from the jshint dist src.

sepe81 commented:
Looks very promising. Have to check this in my test environment during the next days.

sepe81 commented on Gruntfile.js:
```diff
@@ -55,24 +56,36 @@ module.exports = function(grunt) {
+          '<%= build_dir_js %>/angular-chart.min.js': 'angular-chart.js/dist/angular-chart.min.js',
+          '<%= build_dir_js %>/Chart.min.js': 'Chart.js/Chart.min.js',
+          '<%= build_dir_js %>/lodash.js': 'lodash/lodash.js',
+          'build/resources/main/public/css/bootstrap.min.css': 'bootstrap/dist/css/bootstrap.min.css',
```
should consequently be extracted to build_dir_css

sepe81 commented on Gruntfile.js:
```diff
@@ -55,24 +56,36 @@ module.exports = function(grunt) {
+          '<%= build_dir_js %>/angular-route.min.js': 'angular-route/angular-route.min.js',
+          '<%= build_dir_js %>/angular-chart.min.js': 'angular-chart.js/dist/angular-chart.min.js',
+          '<%= build_dir_js %>/Chart.min.js': 'Chart.js/Chart.min.js',
+          '<%= build_dir_js %>/lodash.js': 'lodash/lodash.js',
```
bugfix, change to: '<%= build_dir_js %>/lodash.min.js': 'lodash/lodash.min.js',

sepe81 commented on src/main/resources/public/index.html:
```diff
@@ -3,8 +3,8 @@
-    <link href="bower_components/bootstrap/dist/css/bootstrap.min.css" rel="stylesheet" media="screen">
-    <link href="bower_components/angular-chart.js/dist/angular-chart.css" rel="stylesheet">
+    <link href="css/bootstrap.min.css" rel="stylesheet" media="screen">
+    <link href="css/angular-chart.css" rel="stylesheet">
```
bugfix, change to: <link href="css/angular-chart.min.css" rel="stylesheet">

sepe81 commented:
I commented two bugfixes. 

fonts/glyphicons-halflings-regular.woff2 also needs to be converted...

inpercima commented:
@sepe81 Thank you for checking this PR. I will change and add some interesting ideas.

inpercima commented:
@sepe81 I fixed the bugs. The fonts are now outside of css b/c bootstrap needs this structure. More I extracted the constants from the Gruntfile.js to package.json in sections constants.

sepe81 commented:
Now it works without resource errors! :-)

Did you find any recommendation for putting the grunt constants into package.json or is it selfmade? I'm not sure if we can reuse those constants at other places. Otherwise, if we use the constants just for grunt, why should we put them into another file?

inpercima commented:
It's not selfmade. I found a solution for a problem while copying files using https://github.com/gruntjs/grunt-contrib-copy#process with properties from the package.json. So I desided to use
```javascript
var pkg = grunt.file.readJSON('package.json');

  grunt.initConfig({
    pkg: pkg,
```
In this way I can access from the process function to the package.json and outside of grunt.

Normally, we do not need this in our app. I look up for the link with the solution.

inpercima commented:
@sepe81 To complete this PR I desided to remove the constant section and reuse the constants in grunt. Please check this last changes. :)

inpercima commented:
Hey @sepe81 Can you check this in this week please?

inpercima commented:
@sepe81 please merge this pr.

inpercima commented:
@sepe81 Now you can check and merge.

sepe81 commented:
I removed the unnecessary merge commit and checked it again locally.
